crates/router/src/core/payment_methods/cards.rs

```diff
@@ -456,12 +456,64 @@ pub async fn add_card_hs(
     let store_card_payload =
         call_to_locker_hs(state, &payload, &customer_id, locker_choice).await?;
 
+    // fetch card_bin from db
+
+    let mut card_number = card.card_number.peek().to_owned();
+    let card_isin = req.card.clone().map(|c| c.card_number.get_card_isin());
+    let card = card.clone();
+    let card_detail = card_isin
+        .clone()
+        .async_and_then(|card_isin| async move {
+            state
+                .store
+                .get_card_info(&card_isin)
+                .await
+                .map_err(|error| services::logger::warn!(card_info_error=?error))
+                .ok()
+        })
```

Comment:
This call can be eliminated. The card bin call is already made in the core get_trackers call and the same should be used here.